optuna/integration/chainermn.py

```diff
@@ -233,10 +230,10 @@ def func():
         return self._call_with_mpi(func)
 
     def suggest_categorical(self, name, choices):
-        # type: (str, Sequence[T]) -> T
+        # type: (str, Tuple[Union[float, str], ...]) -> Union[float, str]
```

Comment:
How about supporting `None`, which I think appears in the choice in many use cases?

Reply:
I think that some users use `suggest_categorical` to suggest boolean values (e.g., `suggest_categorical('foo', [False, True])`), so supporting boolean is also useful.

Reply:
I basically pass List(not Tuple) as a `choices` argument like @sile's example. So supporting List is useful to keep the compatibility of type definitions.

Reply:
That makes sense. I simplified the type annotations if you could take another look.